### PR TITLE
[WIP][Bazel][Part 2]: BazelDeps refactor

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Feature: xxxxx
Docs: https://github.com/conan-io/docs/pull/XXXX

Depends on https://github.com/conan-io/conan/pull/14845

### CODE REVIEW

File: conan/tools/google/bazeldeps2.py [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+            # do not sort
+        {% for lib in libs %}
+        ":{{ lib }}_precompiled",
+        {% endfor %}
+        {% for lib in shared_with_interface_libs %}
+        ":{{ lib }}_precompiled",
+        {% endfor %}
+        {% for dep in dependencies %}
+        "@{{ dep }}",
+        {% endfor %}
```
Comment (Neeeflix) [inline conan/tools/google/bazeldeps2.py]: I think indenting this by 4 characters would make the generated file look somewhat nicer

File: conan/tools/google/bazeldeps2.py [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+    cc_import(
+        name = "{{ libname }}_precompiled",
+        interface_library = "{{ lib_path }}",
+        shared_library = "{{ dll_path }}",
+    )
+    {% endfor %}
+
+    cc_library(
+        name = "{{ name }}",
+        {% if headers %}
```
Comment (Neeeflix) [inline conan/tools/google/bazeldeps2.py]:
```suggestion
        {%- if headers %}
```
Doing this everywhere prevents the lines of the generated file from being spread.

File: conan/tools/google/bazeldeps2.py [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+                f" 'build require'. You need to specify a suffix using the "
+                f"'build_context_suffix' attribute at the generator class.")
+
+    # All the requirements
+    host_req = conanfile.dependencies.host
+    build_req = conanfile.dependencies.build  # tool_requires
+    test_req = conanfile.dependencies.test
+
+    # Check if it exists both as require and as build require without a suffix
+    validate_build_requires(host_req, build_req,
```
Comment (Neeeflix) [inline conan/tools/google/bazeldeps2.py]: I am not so deep into the topic, therefore the following question: Is this check also applied to all transitive packages?

**Comment by franramirez688** [comment]
Superseded by https://github.com/conan-io/conan/pull/14959

**Pull request closed without merging**